rs/index/src/posting_list/file.rs

```diff
@@ -88,87 +85,93 @@ impl FileBackedAppendablePostingListStorage {
     pub fn flush(&mut self) -> Result<()> {
         if !self.resident {
             // Flush all mmaps
+            for mmap in self.metadata_mmaps.iter_mut() {
+                mmap.flush()?;
+            }
             for mmap in self.mmaps.iter_mut() {
                 mmap.flush()?;
             }
         }
         Ok(())
     }
 
-    fn new_backing_file(&mut self) -> Result<()> {
-        self.current_backing_id += 1;
+    fn new_backing_file(&mut self, is_metadata: bool) -> Result<()> {
+        let (file_type, mmaps) = if is_metadata {
+            ("metadata", &mut self.metadata_mmaps)
+        } else {
+            ("posting_list", &mut self.mmaps)
+        };
+        let backing_file_size = self.backing_file_size as u64;
+        let base_directory = self.base_directory.clone();
+
+        let file_name = format!("{}/{}.bin.{}", base_directory, file_type, mmaps.len());
+
         let backing_file = OpenOptions::new()
             .read(true)
             .write(true)
             .create(true)
-            .open(format!(
-                "{}/posting_list.bin.{}",
-                self.base_directory, self.current_backing_id
-            ))?;
+            .open(file_name)?;
 
-        backing_file.set_len(self.backing_file_size as u64)?;
+        backing_file.set_len(backing_file_size)?;
+
+        mmaps.push(unsafe { memmap2::MmapMut::map_mut(&backing_file)? });
 
-        self.mmaps
-            .push(unsafe { memmap2::MmapMut::map_mut(&backing_file)? });
-        self.current_offset = 0;
         Ok(())
     }
 
-    fn write_to_current_mmap(&mut self, data: &[u8]) -> Result<()> {
-        let mmap = &mut self.mmaps[self.current_backing_id as usize];
+    fn write_to_current_mmap(&mut self, data: &[u8], is_metadata: bool) -> Result<()> {
```

Comment:
my personal preference is also to have 2 functions: `write_to_metadata_mmaps` and `write_to_posting_list_mmaps`, that internally just calls this function. It's extra, but it makes things easy to read and explicit. I assume the compiler will inline it anyway.